js/coinex.js

```diff
@@ -990,6 +1030,100 @@ module.exports = class coinex extends Exchange {
         return this.parseTrades (trades, market, since, limit);
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeNumber (accountsByType, fromAccount);
+        const toId = this.safeNumber (accountsByType, toAccount);
+        const request = {
+            'coin_type': currency['id'],
+            'amount': this.currencyToPrecision (code, amount),
+        };
+        let method = undefined;
+        if (fromAccount === 'margin' || toAccount === 'margin') {
+            method = 'privatePostMarginTransfer';
+            const symbol = this.safeString (params, 'symbol');
+            if (symbol === undefined) {
+                throw new ExchangeError ('You must define in params what market you are transferring. e.g. {symbol: ETH/BTC}');
```

Comment:
Let's use this uniform exception message format:

```JavaScript
throw new ArgumentsRequired (thid.id + ' transfer() requires a symbol parameter')
throw new ArgumentsRequired (thid.id + ' createOrder() requires a price argument')
throw new ArgumentsRequired (thid.id + ' fetchTransfers() requires an account_id parameter')
...
```

Also, let's make sure we don't forget to omit `symbol` from `params` before sending out the request.

Logically, we must support the exchange-specific parameter name `market` for the marketId as well as the unified parameter name `symbol` for unified market symbols.

1. Check if `market` is specified in `params`.
2. If `params['market']` is specified, then it contains the market id.
3. If it's not specified, check if `symbol` is specified.
4. If `params['symbol']` is specified, convert the unified symbol to the market id and omit the `symbol` from `params`, otherwise it will be sent out to the exchange (that does not support the `symbol` parameter) because the request is extended with `params` later upon sending out the request.

```JavaScript
if (fromAccount === 'margin' || toAccount === 'margin') {
    const marketId = this.safeString (params, 'market');
    // if not specified, check the symbol parameter as well
    // otherwise, if specified, it will be sent automatically
    // because the request is extended with params before sending it out
    if (marketId === undefined) {
        const symbol = this.safeString (params, 'symbol');
        if (symbol === undefined) {
            throw new ExchangeError (this.id + ' transfer() requires an exchange-specific market parameter or a unified symbol parameter');
        } else {
            params = this.omit (params, 'symbol');
            const market = this.market (symbol);
            request['market'] = market['id'];
            // ...
        }
    }
    // ...
```